src/transformStateWithClones.js

```diff
@@ -8,6 +8,38 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const stateCopy = { ...state };
+  const results = [];
+
+  for (const act of actions) {
+    const { type, extraData, keysToRemove } = act;
```

Comment:
You can destructure it right in loop
```suggestion
  for (const { type, extraData, keysToRemove } of actions) {
```